syntax = "proto3";
package transport;

import "google/protobuf/timestamp.proto";
import "google/protobuf/any.proto";

message Envelope {
  string topic = 1;
  uint64 seq = 2;
  google.protobuf.Any payload = 3;
  google.protobuf.Timestamp ts = 4;
  bytes uuid = 5; // 16-byte binary UUID
}

message Heartbeat {
  string node_id = 1;
  string zmq_addr = 2;
  map<string, uint64> last_seq = 3;
}

message FetchRequest { string topic = 1; uint64 from = 2; uint64 to = 3; }
message FetchResponse { repeated Envelope envelopes = 1; }
